test/libyul/yulOptimizerTests/fullSuite/medium.yul

```diff
@@ -9,6 +9,9 @@
     pop(allocate(0x20))
     let x := allocate(0x40)
     mstore(array_index_access(x, 3), 2)
+    if 0 {
```

Comment:
Please add a more extensive test that shows that the full suite can also simplify the conditions accordingly.

Reply:
There may be a problem in some cases, since we may end up with
```
let _1 := 1
if _1
{
 ...
}
```
And the structural simplifier doesn't know that ``_1`` is ``1`` - I'll check how best to avoid that.

Reply:
I would say either use the `DataFlowAnalyzer` or use the rematerializer.